openssh/server/public_key.sls

```diff
@@ -39,6 +39,14 @@
 
 {%- endif %}
 
+{%- else %}
+
+{%- if user.get('purge', False) %}
```

Comment:
Convention is to used `enabled` keyword which if is false instructs states to remove given resources.